File: Examples/CSharp/WorkingWithProjects/GetNumberOfPagesForViews.cs
namespace Aspose.Tasks.Examples.CSharp.WorkingWithProjects
{
    using System;

    using Aspose.Tasks.Visualization;

    internal class GetNumberOfPagesForViews
    {
        public static void Run()
        {
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod());
            
            //ExStart:GetNumberOfPagesForViews
            //ExFor: Project.GetPageCount(PresentationFormat,Timescale)
            //ExSummary: Shows how to get count of pages by presentation format and timescale.
            // Read the source Project
            var project = new Project(dataDir + "GetNumberOfPagesForViews.mpp");

            // Get number of pages,  Months and  ThirdsOfMonths
            Console.WriteLine("Number of Pages = '{0}'", project.GetPageCount(PresentationFormat.ResourceUsage, Timescale.Days));
            Console.WriteLine("Number of Pages = '{0}'", project.GetPageCount(PresentationFormat.ResourceUsage, Timescale.Months));
            Console.WriteLine("Number of Pages = '{0}'", project.GetPageCount(PresentationFormat.ResourceUsage, Timescale.ThirdsOfMonths));
            //ExEnd:GetNumberOfPagesForViews
        }
    }
}

File: Examples/VisualBasic/WorkingWithProjects/GetNumberOfPagesForViews.vb
Imports System
Imports Aspose.Tasks.Visualization

Namespace WorkingWithProjects
    Public Class GetNumberOfPagesForViews
        Public Shared Sub Run()
            ' ExStart: GetNumberOfPagesForViews
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            '  Read the source Project Get number of pages, Months and ThirdsOfMonths
            Dim project As New Project(dataDir & "GetNumberOfPagesForViews.mpp")
            Console.WriteLine(String.Format("Number of Pages = '{0}'", project.GetPageCount(PresentationFormat.ResourceUsage, Timescale.Days)))
            Console.WriteLine(String.Format("Number of Pages = '{0}'", project.GetPageCount(PresentationFormat.ResourceUsage, Timescale.Months)))
            Console.WriteLine(String.Format("Number of Pages = '{0}'", project.GetPageCount(PresentationFormat.ResourceUsage, Timescale.ThirdsOfMonths)))
            ' ExEnd: GetNumberOfPagesForViews
        End Sub
    End Class
End Namespace